packages/foam-core/src/markdown-provider.ts

```diff
@@ -145,8 +146,17 @@ export function createMarkdownReferences(
   return graph
     .getForwardLinks(noteId)
     .map(link => {
-      const target = graph.getNote(link.to);
-
+      let target = graph.getNote(link.to);
+      // if we don't find the target by ID we search the graph by slug
+      if (!target) {
+        const candidates = graph.getNotes({ slug: link.link.slug });
```

Comment:
we are now explicitly finding out how many notes exist with the given slug and can act on it (atm logging to console for consistency with the rest of the reporting in this fn)